src/Microsoft.PowerShell.Commands.Management/commands/management/TestConnectionCommand.cs

```diff
@@ -862,7 +862,7 @@ private IPHostEntry GetCancellableHostEntry(string targetNameOrAddress)
         // Creates and fills a send buffer. This follows the ping.exe and CoreFX model.
         private static byte[] GetSendBuffer(int bufferSize)
         {
-            if (bufferSize == DefaultSendBufferSize && s_DefaultSendBuffer != null)
+            if (bufferSize == DefaultSendBufferSize)
             {
                 return s_DefaultSendBuffer;
             }
```

Comment:
> GetSendBuffer() is still needed when using non-default buffer size.

In that case, it seems wrong to return an empty array just because the `buffersize` is the same as the default buffer size.

Reply:
The change in .NET is that the default buffer is 32 bytes, but has to either not be specified (with an overload that doesn't provide buffer) or has to be an empty array (anything else will throw on Linux).

The alternative is as @iSazonov suggests to add code to check if Linux and if root if `-BufferSize` is specified even though the error message from .NET already tells you that you need to be elevated if a non-standard buffer is used on Linux.

Reply:
Okay, so when `BufferSize == 32`, no matter it's the default value in play or the user explicitly set `-BufferSize 32`, we will use an empty array for `buffer`. Otherwise, we fill the array with auto-generated bytes. Looks good to me.